### PR TITLE
chore: expect specific errors in apps/transfer

### PR DESCRIPTION
**Pull request by bznein**
## Description

ref: #7175

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/transfer/keeper/grpc_query_test.go [hunk 1]
```diff
@@ -181,16 +182,21 @@ func (suite *KeeperTestSuite) TestQueryDenomHash() {
...
+			"success",
+			func() {},
+			nil,
+		},
 		{
 			"invalid trace",
 			func() {
 				req = &types.QueryDenomHashRequest{
-					Trace: "transfer/channelToA/transfer/",
+					Trace: "transfer%%/channel-1/transfer/channel-1/uatom",
```
Comment (bznein) [inline modules/apps/transfer/keeper/grpc_query_test.go:196]: This test was not triggering the "invalid trace" error but the "denom not found one". Thanks @crodriguezvega for the help!!

File: modules/apps/transfer/keeper/grpc_query_test.go [hunk 2]
```diff
@@ -199,12 +205,7 @@ func (suite *KeeperTestSuite) TestQueryDenomHash() {
 					Trace: "transfer/channelToC/uatom",
 				}
 			},
-			false,
-		},
-		{
-			"success",
-			func() {},
-			true,
```
Comment (bznein) [inline modules/apps/transfer/keeper/grpc_query_test.go:207]: moved on top

File: testing/utils.go [hunk 3]
```diff
@@ -81,3 +83,12 @@ func UnmarshalMsgResponses(cdc codec.Codec, data []byte, msgs ...codec.ProtoMars
 
 	return nil
 }
+
```
Comment (bznein) [inline testing/utils.go:86]: This was also added in another in-flight PR. There should be no merge conflicts, but I'll take care of them if they arise :)

File: modules/apps/transfer/keeper/grpc_query_test.go [hunk 4]
```diff
@@ -91,12 +92,12 @@ func (suite *KeeperTestSuite) TestQueryDenom() {
...
 			res, err := suite.chainA.GetSimApp().TransferKeeper.Denom(ctx, req)
 
-			if tc.expPass {
+			if tc.expErr == nil {
 				suite.Require().NoError(err)
 				suite.Require().NotNil(res)
 				suite.Require().Equal(&expDenom, res.Denom)
 			} else {
-				suite.Require().Error(err)
+				ibctesting.RequireErrorIsOrContains(suite.T(), err, tc.expErr, err.Error())
```
Comment (crodriguezvega) [inline modules/apps/transfer/keeper/grpc_query_test.go:100]: Maybe in a separate PR we should update other tests where this function could be used... More refactoring of tests, yay!!

**Review by crodriguezvega — APPROVED** [review]
😍

File: modules/apps/transfer/keeper/grpc_query_test.go (hunk 4 above)
Comment (bznein) [reply modules/apps/transfer/keeper/grpc_query_test.go:100]: Definitely! But I think after these I'll take a break from refactoring tests :D

File: modules/apps/transfer/keeper/grpc_query_test.go (hunk 1 above)
Comment (damiannolan) [reply modules/apps/transfer/keeper/grpc_query_test.go:196]: Is the %% what triggers invalid trace here?

**Review by damiannolan — APPROVED** [review]
Nice one, thanks @bznein!

File: modules/apps/transfer/keeper/grpc_query_test.go (hunk 1 above)
Comment (bznein) [reply modules/apps/transfer/keeper/grpc_query_test.go:196]: To my understanding yes, but this is all @crodriguezvega!

Comment (crodriguezvega) [reply modules/apps/transfer/keeper/grpc_query_test.go:196]: Yes, correct. The `%%` triggers invalid trace error.

**Pull request merged**